src/renderers/dom/client/ReactBrowserEventEmitter.js

```diff
@@ -296,6 +297,16 @@ var ReactBrowserEventEmitter = assign({}, ReactEventEmitterMixin, {
               mountAt
             );
             ReactBrowserEventEmitter.ReactEventListener.trapCapturedEvent(
+              topLevelTypes.topFocusIn,
```

Comment:
This isn't feature detected. Note:

https://developer.mozilla.org/en-US/docs/Web/Events/focusin

Firefox (Gecko): Not supported.

This would need to be normalized, but if we do then it works exactly like focus/blur. So I think we should revert.